google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/FileConfirmationDialog.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gct.idea.appengine.cloud;
+
+import com.google.gct.idea.util.GctBundle;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.util.SystemInfo;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+/**
+ * A dialog warning for, e.g., overwriting existing app.yaml and Dockerfile configuration files
+ * when user selects a destination folder to auto-generate them.
+ */
+public class FileConfirmationDialog extends DialogWrapper {
+  public enum DialogType { CONFIRM_OVERWRITE, CONFIRM_CREATE_DIR, NOT_DIRECTORY_ERROR };
+
+  private JPanel rootPanel;
+  private JLabel warningLabel;
+  private JLabel pathDisplay;
+
+  private DialogType dialogType;
+
+  public FileConfirmationDialog(
+      @Nullable Project project, DialogType dialogType, @NotNull File targetPath) {
+    super(project);
+
+    this.dialogType = dialogType;
+
+    switch (dialogType) {
+      case CONFIRM_OVERWRITE:
+        warningLabel.setText(GctBundle.message("appengine.flex.config.destination.overwrite"));
+        break;
+
+      case CONFIRM_CREATE_DIR:
+        warningLabel.setText(GctBundle.message("appengine.flex.config.destination.create.dir"));
+        break;
+
+      case NOT_DIRECTORY_ERROR:
+        warningLabel.setText(GctBundle.message("appengine.flex.config.destination.not.directory"));
+        break;
+
+      default:
+        throw new AssertionError();
+    }
+    pathDisplay.setText(targetPath.getAbsolutePath());
+
+    init();
+  }
+
+  @Nullable
+  @Override
+  protected JComponent createCenterPanel() {
+    return rootPanel;
+  }
+
+  @NotNull
+  @Override
+  protected Action[] createActions() {
+    if (dialogType == DialogType.NOT_DIRECTORY_ERROR) {
+      return new Action[]{ getOKAction() };
+    } else {
+      if (SystemInfo.isMac) {
```

Comment:
Is this a Mac UI thing?  I think we should let the platform handle matching the host OS. If we write point solutions like this we'll create inconsistencies.